packages/smithy-client/src/extensions/retry.ts

```diff
@@ -0,0 +1,27 @@
+import { Provider, RetryStrategy, RetryStrategyConfiguration, RetryStrategyV2 } from "@smithy/types";
+
+/**
+ * @internal
+ */
+export const getRetryConfiguration = (
+  runtimeConfig: Partial<{ retryStrategy: Provider<RetryStrategyV2 | RetryStrategy> }>
+) => {
+  return {
+    _retryStrategy: runtimeConfig.retryStrategy!,
+    setRetryStrategy(retryStrategy: Provider<RetryStrategyV2 | RetryStrategy>): void {
+      this._retryStrategy = retryStrategy;
```

Comment:
what is the scope of this setter? is it supposed to mutate the retryStrategy of a client for all future requests?